hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java

```diff
@@ -69,26 +69,9 @@ public void enrichEvent(StructuredTrace trace, Event event) {
   private Optional<String> getUserAgent(Event event) {
     Protocol protocol = EnrichedSpanUtils.getProtocol(event);
     if (Protocol.PROTOCOL_HTTP == protocol || Protocol.PROTOCOL_HTTPS == protocol) {
-      if (event.getHttp() != null && event.getHttp().getRequest() != null) {
-        // prefer user agent from headers
-        if (event.getHttp().getRequest().getHeaders() != null
-            && !StringUtils.isEmpty(event.getHttp().getRequest().getHeaders().getUserAgent())) {
-          return Optional.of(event.getHttp().getRequest().getHeaders().getUserAgent());
-        }
-
-        // fallback to user agent on the request
-        if (!StringUtils.isEmpty(event.getHttp().getRequest().getUserAgent())) {
-          return Optional.of(event.getHttp().getRequest().getUserAgent());
-        }
-      }
+      return HttpSemanticConventionUtils.getHttpUserAgent(event);
```

Comment:
HttpSemanticConventionUtils.getHttpUserAgent are we taking care of getting userAgent from (event.getHttp().getRequest().getHeaders() as well?

Reply:
So I had a doubt in this. Since this method is only being used here, can we change the order of attributes in the USER_AGENT_ATTRIBUTES to make sure the one accounting for header is at first position. Or add that thing as a separate condition

Reply:
Can you make it two method, one which gets userAgent from request and other one gets its from headers

If you club, some other consumer might be interested in the reverse order, so I guess better to keep it separate